library/ratelimiting/shouldRateLimitRequest.ts

```diff
@@ -57,13 +57,14 @@ export function shouldRateLimitRequest(
   const { maxRequests, windowSizeInMS } = endpoint.rateLimiting;
 
   if (context.user) {
+    let key = `${endpoint.method}:${endpoint.route}:user:${context.user.id}`;
+    if (context.user.rateLimitGroup) {
+      key = `${endpoint.method}:${endpoint.route}:group:${context.user.rateLimitGroup}`;
+    }
+
     const allowed = agent
       .getRateLimiter()
-      .isAllowed(
-        `${endpoint.method}:${endpoint.route}:user:${context.user.id}`,
-        windowSizeInMS,
-        maxRequests
-      );
+      .isAllowed(key, windowSizeInMS, maxRequests);
 
     if (!allowed) {
       return { block: true, trigger: "user" };
```

Comment:
Not sure if we should change trigger? Would also be a change for users using a custom middleware.

Reply:
It would be nice that the `trigger` would be `group` but indeed, it might break custom middlewares... given the security context, safer to keep it like this?